Framework/Core/src/InputSpec.cxx

```diff
@@ -89,10 +88,43 @@ InputSpec::InputSpec(std::string binding_,
 {
 }
 
+InputSpec InputSpec::fromString(std::string s)
```

Comment:
Did you check some of the utilities we already have in `Framework/Core/include/Framework/DataDescriptorQueryBuilder.h`?
If you really need something else, could you move this code there in any case?

Reply:
For the source table tracking I have a special format, as we also want to track if the particular input is enabled by a process switch. While I can use that for parsing the substring, it is easier and faster to just apply regex. My code is very specific for InputSpec metadata, though, DataDescriptorQueryBuilder has a different purpose.

Reply:
I would use either DataDescriptorQueryBuilder or DataSpecUtils. How to parse something to produce an InputSpec is not something which is part of the "essence" of being an `InputSpec`, so it should not be in `InputSpec.h`.